framework/include/image_data.hpp

```diff
@@ -0,0 +1,282 @@
+			: mPaths(aPaths), mHDR(aLoadHdrIfPossible), msRGB(aLoadSrgbIfApplicable), mFlip(aFlip), mPreferredNumberOfTextureComponents(aPreferredNumberOfTextureComponents)
+		{
+		}
+
+		// Note that boolean flags are not applicable in all cases; stb_image converts image data to or from float format, but other image loading libraries, like GLI, do not;
+		// Likewise, sRGB is ignored for GLI loader, and it may not be possible to flip images on loading (e.g. compressed textures)
+		static std::unique_ptr<image_data_implementor> load_image_data_from_file(const std::string& aPath, const bool aLoadHdrIfPossible = true, const bool aLoadSrgbIfApplicable = true, const bool aFlip = true, const int aPreferredNumberOfTextureComponents = 4);
+		
+		// for cubemaps loaded from six individual files
+		// Order of faces +X, -X, +Y, -Y, +Z, -Z
+		static std::unique_ptr<image_data_implementor> load_image_data_from_file(const std::vector<std::string>& aPaths, const bool aLoadHdrIfPossible = true, const bool aLoadSrgbIfApplicable = true, const bool aFlip = true, const int aPreferredNumberOfTextureComponents = 4);
+		
+		std::vector<std::string> mPaths;
+
+		bool mHDR;
+		bool msRGB;
+		// if image should be flipped vertically when loaded, if possible
+		bool mFlip;
+		int mPreferredNumberOfTextureComponents;
+	};
+
+	// base class of implementor of image_data type in bridge pattern
+	class image_data_implementor : public image_data_interface
+	{
+	public:
+		// Default implementations for subclasses that don't support these optional features
+
+		// Mipmap levels; 1 if no Mipmapping, 0 if Mipmaps should be created after loading
+		virtual uint32_t levels() const 
+		{
+			return 1;
+		}
+
+		// array layers, for texture arrays
+		virtual uint32_t layers() const
+		{
+			return 1;
+		}
+
+		// faces in cubemap, must be 6 for cubemaps, 1 for anything else
+		virtual uint32_t faces() const
+		{
+			return 1;
+		}
+
+		// TODO: make protected or remove?
+		virtual bool can_flip() const
+		{
+			return false;
+		}
+
+		virtual bool is_hdr() const
+		{
+			return false;
+		}
+
+	protected:
+		explicit image_data_implementor(const std::string& aPath, const bool aLoadHdrIfPossible = false, const bool aLoadSrgbIfApplicable = false, const bool aFlip = false, const int aPreferredNumberOfTextureComponents = 4)
+			: image_data_interface(aPath, aLoadHdrIfPossible, aLoadSrgbIfApplicable, aFlip, aPreferredNumberOfTextureComponents)
+		{
+		}
+
+		explicit image_data_implementor(const std::vector<std::string>& aPaths, const bool aLoadHdrIfPossible = false, const bool aLoadSrgbIfApplicable = false, const bool aFlip = false, const int aPreferredNumberOfTextureComponents = 4)
+			: image_data_interface(aPaths, aLoadHdrIfPossible, aLoadSrgbIfApplicable, aFlip, aPreferredNumberOfTextureComponents)
+		{
+		}
+	};
+
+	// base class of abstraction in bridge pattern
+	class image_data_t : public image_data_interface
```

Comment:
```suggestion
	class image_data : public image_data_interface
```
`image_data` still uses the pimpl idiom and it still uses a `unique_ptr` internally to store the data, _which is fine_ since this class represents image _data_! It does not represent an image resource--i.e. in the sense of a "GPU resource type". 

The point is: If a `unique_ptr` is used anyways (there are no other options to using a `unique_ptr` the way the class is implemented... _which is fine_!), wrapping it in a `owning_resource` does not help anything! It just makes things more complicated. I.e. let's just remove the `owning_resource` wrapper and be good with it!